### PR TITLE
support extend in config

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: docs/designs/config.md [hunk 1]
```diff
@@ -59,7 +59,7 @@
 
 ``` yml
 name: Azure.azure-documents
-default: msdocs.yml
+extend: msdocs.yml
```
Comment (yufeih) [inline docs/designs/config.md]: where do you think we put 'msdocs.yml'?

Comment (superyyrrzz) [reply docs/designs/config.md]: It can be in a separate repo, restored in current config.

**Comment by superyyrrzz** [comment]
the behavior of `.eslintrc.js`'s `extends`:
1. current config takes precedence over `extends` config
2. latter config takes precedence over former config

will change accordingly

File: src/docfx/config/Config.cs [hunk 2]
```diff
@@ -107,6 +102,52 @@ private static Config LoadCore(string configPath, CommandLineOptions options = n
...
         }
 
+        private static JObject LoadOriginalConfigObject(string configPath, List<string> parents)
+        {
+            var result = YamlUtility.Deserialize<JObject>(File.ReadAllText(configPath)) ?? new JObject();
+            if (!result.TryGetValue(ConfigConstants.Extend, out var objExtend))
+                return result;
+
+            if (parents.Contains(configPath))
+                throw Errors.CircularReference(configPath, parents);
```
[the remark this answers is not in the document]
Comment (superyyrrzz) [reply src/docfx/config/Config.cs]: Agree. Removed it.

File: docs/designs/config.md [hunk 3]
```diff
@@ -118,7 +118,10 @@ Configs is grouped by scenario:
 * `contribution`: to config whether to enable contribution, and where to config.
 * `contrtibutors`: to config which contributors to include or exclude.
 * `gitDependencies`: to config CRRs.
-Extension config is supported to share default config for all OPS repos, by `defaults: msdocs.yml`. Current config has higher priority than extension config when conflicts.
+Extension config is supported to share default config for all OPS repos, by `extend: msdocs.yml`. Current config has higher priority than extension config when conflicts. Extension config also supports array, whose latter item has higher priority.
+> [!NOTE]
+>
+> `dependencies` doesn't support `extend`, as we may import extending configs from dependent repos.
```
Comment (yufeih) [inline docs/designs/config.md]: having dependency repos in a shared location is a valid scenario, we may want to put localization config for all repos in a central repo,

Comment (superyyrrzz) [reply docs/designs/config.md]: Yes, this should be supported. Changed the wording to make it clear...

**Pull request merged**